Source/WebCore/worklets/PaintWorkletGlobalScope.cpp

```diff
@@ -42,7 +42,7 @@ WTF_MAKE_TZONE_OR_ISO_ALLOCATED_IMPL(PaintWorkletGlobalScope);
 RefPtr<PaintWorkletGlobalScope> PaintWorkletGlobalScope::tryCreate(Document& document, ScriptSourceCode&& code)
 {
     RefPtr<VM> vm = VM::tryCreate();
-    if (!vm)
+    if (!vm || !document.sessionID())
```

Comment:
Checking for `document.sessionID()` is likely not the right thing to do here. How about `!document.frame()` instead?

Reply:
I don't think we really care either way. We're not shipping PaintWorklet anytime soon, and the spec doesn't mention browsing context at all so it won't be spec compliant anyway. It's likely a spec bug.

Reply:
I care. We check that documents have frames all the time in Web APIs. Checking that a document has a session ID? no. The only reason the document doesn't have a sessionID is because it doesn't have a frame likely.

Reply:
> I care. We check that documents have frames all the time in Web APIs. Checking that a document has a session ID? no. The only reason the document doesn't have a sessionID is because it doesn't have a frame likely.

But there is no basis in the spec to do that.

Reply:
I am unable to verify due to WTR crashes on recent ASAN builds. If you are sure, I can update the PR.

Reply:
@pratiksha-99 : For expediency, you should just go ahead & do that. But looking at the code more closely, it's probably not the right place to add a check because returning nullptr results in OutOfMemoryError to be emitted in `PaintWorklet::addModule` in `DOMCSSPaintWorklet.cpp`. That's certainly not the right type of error to throw. We should add a check for `frame()` returning null in `PaintWorklet::addModule` like this (we should use `RefPtr` in place of `auto*` for document):
```cpp
void PaintWorklet::addModule(const String& moduleURL, WorkletOptions&&, DOMPromiseDeferred<void>&& promise)
{
    RefPtr document = this->document();
    if (!document) {
        promise.reject(Exception { ExceptionCode::InvalidStateError, "This frame is detached"_s });
        return;
    }
    if (!document->hasBrowsingContext()) {
        promise.reject(Exception { ExceptionCode::InvalidStateError, "This document does not have a browsing context"_s });
        return;
    }
```